test/e2e/port_forward.go

```diff
@@ -7,19 +7,22 @@ import (
 	"net/url"
 	"strings"
 
+	"github.com/stretchr/testify/require"
 	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/portforward"
 	"k8s.io/client-go/transport/spdy"
 )
 
 // CreatePortForward listens for local connections and forwards them to a remote pod
-func CreatePortForward(namespace, pod string, ports []string, kConfig *rest.Config) (*portforward.PortForwarder, chan struct{}, error) {
+func CreatePortForward(namespace, podNamePrefix, containsImage string, ports []string, kConfig *rest.Config) (*portforward.PortForwarder, chan struct{}) {
 	roundTripper, upgrader, err := spdy.RoundTripperFor(kConfig)
 	if err != nil {
-		return nil, nil, err
+		printTestStackTrace()
```

Comment:
Thought this may be a mistake until saw the comment in the function. Can you include an example of the output to show why it is easier to read?

Reply:
For example, if we fail in GetPod printTestStackTrace() will print something like this:

        /Users/kearls/go/src/github.com/jaegertracing/jaeger-operator/test/e2e/utils.go#60
	/Users/kearls/go/src/github.com/jaegertracing/jaeger-operator/test/e2e/port_forward.go#24
	/Users/kearls/go/src/github.com/jaegertracing/jaeger-operator/test/e2e/daemonset_test.go#108

While debug.printStack will print this:

goroutine 603 [running]:
runtime/debug.Stack(0x2c, 0x0, 0x0)
	/usr/local/Cellar/go/1.12.6/libexec/src/runtime/debug/stack.go:24 +0x9d
runtime/debug.PrintStack()
	/usr/local/Cellar/go/1.12.6/libexec/src/runtime/debug/stack.go:16 +0x22
github.com/jaegertracing/jaeger-operator/test/e2e.GetPod(0xc0006326a0, 0x19, 0x21ea673, 0x9, 0x220fa61, 0x18, 0x2475380, 0xc0002a3680, 0x0, 0x0, ...)
	/Users/kearls/go/src/github.com/jaegertracing/jaeger-operator/test/e2e/utils.go:62 +0x22e
github.com/jaegertracing/jaeger-operator/test/e2e.CreatePortForward(0xc0006326a0, 0x19, 0x21ea673, 0x9, 0x220fa61, 0x18, 0xc0001aba80, 0x1, 0x1, 0xc00000c3c0, ...)
	/Users/kearls/go/src/github.com/jaegertracing/jaeger-operator/test/e2e/port_forward.go:24 +0xf5
github.com/jaegertracing/jaeger-operator/test/e2e.(*DaemonSetTestSuite).TestDaemonSet(0xc000149560)
	/Users/kearls/go/src/github.com/jaegertracing/jaeger-operator/test/e2e/daemonset_test.go:108 +0x9d4
reflect.Value.call(0xc0008d7320, 0xc0002a04e8, 0x13, 0x21df509, 0x4, 0xc00095df30, 0x1, 0x1, 0xc0008c4e68, 0x78, ...)
	/usr/local/Cellar/go/1.12.6/libexec/src/reflect/value.go:447 +0x461
reflect.Value.Call(0xc0008d7320, 0xc0002a04e8, 0x13, 0xc0008c4f30, 0x1, 0x1, 0x1f06ce3, 0xd, 0x0)
	/usr/local/Cellar/go/1.12.6/libexec/src/reflect/value.go:308 +0xa4
github.com/jaegertracing/jaeger-operator/vendor/github.com/stretchr/testify/suite.Run.func2(0xc0008dc000)
	/Users/kearls/go/src/github.com/jaegertracing/jaeger-operator/vendor/github.com/stretchr/testify/suite/suite.go:126 +0x2fb
testing.tRunner(0xc0008dc000, 0xc00023d400)
	/usr/local/Cellar/go/1.12.6/libexec/src/testing/testing.go:865 +0xc0
created by testing.(*T).Run
	/usr/local/Cellar/go/1.12.6/libexec/src/testing/testing.go:916 +0x35a